common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -658,14 +658,20 @@ size_t base_wlan_hal_dwpal::dwpal_nl_cmd_get(const std::string &ifname, unsigned
         return nl_handler_cb_wrapper(ifname, event, subevent, len, data);
     };
 
-    //parsing will be done in callback func
-    if (dwpal_driver_nl_msg_get(m_dwpal_nl_ctx, DWPAL_NL_SOLICITED_EVENT, nl_handler_cb, NULL) ==
-        DWPAL_FAILURE) {
-        LOG(ERROR) << " dwpal_driver_nl_msg_get failed,"
-                   << " ctx=" << m_dwpal_nl_ctx;
-        return data_size;
+    // Reply from driver could be delayed so we need
+    // to continue polling the answer until with valid size.
+    // This is a blocking function so we don't want to stay here
+    // more than 1 sec.
+    auto timeout = std::chrono::steady_clock::now() + std::chrono::seconds(1);
+    while ((data_size == 0) && std::chrono::steady_clock::now() < timeout) {
+        //parsing will be done in callback func
+        if (dwpal_driver_nl_msg_get(m_dwpal_nl_ctx, DWPAL_NL_SOLICITED_EVENT, nl_handler_cb,
```

Comment:
Since `dwpal_driver_nl_msg_get` is blocking, theoretically it can block the calling thread for more than 1 second.
In practice, the response usually arrives much faster. Perhaps the comment should be updated.

Reply:
@vitalybu 's explanation is excellent - please use it in the comment.

Reply:
clearly that is not what i meant you to understand from the comment,

dwpal_nl_cmd_get is blocking , inside i run   dwpal_driver_nl_msg_get in a loop with 1 sec TO 
I'll change the comment....